src/main/java/com/pnu/ordermanagementapp/product/ProductRepository.java

```diff
@@ -0,0 +1,40 @@
+package com.pnu.ordermanagementapp.product;
+
+import com.pnu.ordermanagementapp.model.Product;
+import org.springframework.data.domain.Example;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.PagingAndSortingRepository;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface ProductRepository extends JpaRepository<Product, Long>, PagingAndSortingRepository<Product, Long> {
+
```

Comment:
For what two implementations? I believe only `PagingAndSortingRepository<Product, Long>` provides the same alone.

Reply:
`JpaRepository` alone provide both.
However, it's out of the scope of this story to refactor such kind of stuff.
Only mininal required refactoring was performed, just to be able to integrate security into our app.
We have another task for refactoring.
@volodymyrvynnyk take a look at this comment, it's one of the things for you to refactor in scope of refactoring story.